app/src/main/java/org/oppia/app/player/state/itemviewmodel/NumericInputViewModel.kt

```diff
@@ -17,4 +81,57 @@ class NumericInputViewModel : StateItemViewModel(ViewType.NUMERIC_INPUT_INTERACT
     }
     return userAnswerBuilder.build()
   }
+
+  /**
+   * Returns a [NumericInpurParsingError] for obvious incorrect number formatting issues for the specified raw text, or
+   * [NumericInpurParsingError.VALID] if not such issues are found.
+   *
+   * Note that this method returning a valid result does not guarantee the text is a valid number--
+   * [getSubmitTimeError] should be used for that, instead. This method is meant to be used as a quick sanity check for
+   * general validity, not for definite correctness.
+   */
+  fun getRealTimeAnswerError(text: String): NumericInpurParsingError {
+    val normalized = text.normalizeWhitespace()
+    return when {
+      !normalized.matches(invalidCharsRegex) -> NumericInpurParsingError.INVALID_CHARS
+      normalized.startsWith(".") -> NumericInpurParsingError.STARTING_WITH_FLOATING_POINT
+      normalized.count { it == '.' } > 1 -> NumericInpurParsingError.INVALID_FORMAT
+      normalized.lastIndexOf('-') > 0 -> NumericInpurParsingError.INVALID_FORMAT
+      else -> NumericInpurParsingError.VALID
+    }
+  }
+
+  /**
+   * Returns a [NumericInpurParsingError] for the specified text input if it's an invalid number, or
+   * [NumericInpurParsingError.VALID] if no issues are found. Note that a valid number returned by this method is guaranteed
+   * to be parsed correctly.
+   *
+   * This method should only be used when a user tries submitting an answer. Real-time error detection should be done
+   * using [getRealTimeAnswerError], instead.
+   */
+  fun getSubmitTimeError(text: String): NumericInpurParsingError {
+    if (invalidCharsLengthRegex.find(text) != null) {
+      return NumericInpurParsingError.NUMBER_TOO_LONG
+    }
+    return try {
+      text.toDouble()
+      NumericInpurParsingError.VALID
+    } catch (e: Exception) {
+      NumericInpurParsingError.INVALID_FORMAT
+    }
+  }
+
+  /** Enum to store the errors of [NumericInputInteractionView]. */
+  enum class NumericInpurParsingError(@StringRes private var error: Int?) {
```

Comment:
Typo error make it NumericInputParsingError().

Reply:
Done.